extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -185,36 +239,47 @@ public Table listTablesAsTable(@NotNull final Namespace namespace) {
         final Map<String, ColumnSource<?>> columnSourceMap = new LinkedHashMap<>();
 
         // Create the column source(s)
-        final WritableColumnSource<String> namespaceColumn =
-                ArrayBackedColumnSource.getMemoryColumnSource(size, String.class, null);
-        columnSourceMap.put("namespace", namespaceColumn);
+        final String[] namespaceArr = new String[(int) size];
+        columnSourceMap.put("namespace", InMemoryColumnSource.getImmutableMemoryColumnSource(namespaceArr));
 
-        final WritableColumnSource<String> tableColumn =
-                ArrayBackedColumnSource.getMemoryColumnSource(size, String.class, null);
-        columnSourceMap.put("table_name", tableColumn);
+        final String[] tableNameArr = new String[(int) size];
+        columnSourceMap.put("table_name", InMemoryColumnSource.getImmutableMemoryColumnSource(tableNameArr));
 
-        final WritableColumnSource<Object> objectColumn =
-                ArrayBackedColumnSource.getMemoryColumnSource(size, Object.class, TableIdentifier.class);
-        columnSourceMap.put("table_identifier_object", objectColumn);
+        final TableIdentifier[] tableIdentifierArr = new TableIdentifier[(int) size];
+        columnSourceMap.put("table_identifier_object",
+                InMemoryColumnSource.getImmutableMemoryColumnSource(tableIdentifierArr, Namespace.class, null));
```

Comment:
Data type was wrong.
```suggestion
                InMemoryColumnSource.getImmutableMemoryColumnSource(tableIdentifierArr));
```